### DISCUSSION TITLE
feat: add support for LoudMax

### DISCUSSION THREAD

**Discussion by bangbangsheshotmedown** [wwmm/easyeffects#3486] [Ideas]
Great plugin for loudness equalization

Perfect for people without a phd in sound engineering

https://loudmax.blogspot.com/2023/?m=1

**Comment by violetmage** [comment]
According to the website, loudmax is freeware, and thus its license is not compatible with the license that easyeffects is published under (GPLv3+).

Because loudmax's license is incompatible, it cannot be legally included as a part of easyeffects.

If you really need to use loudmax outside of a DAW, [carla](https://kx.studio/Applications:Carla) is probably what you want.

**Reply by wwmm** [reply]
Is the license still a problem if the plugin is not shipped together with EasyEffects code? In any case even if somehow the license issue can be solved there will still be the question about whether it is necessary to add a plugin that does things we can already do with the plugins we currently use. We already have a maximizer as well as a limiter and a loudness corrector. Adding several plugins that do the same thing were never our approach.

**Reply by bangbangsheshotmedown** [reply]
none of the existing ones are as easy as this:

![image](https://github.com/user-attachments/assets/3cadefcf-cb25-400b-8ead-72870b0cc3d8)

so i think there is value in having one that just works with minimal setup

**Reply by bangbangsheshotmedown** [reply]
> According to the website, loudmax is freeware, and thus its license is not compatible with the license that easyeffects is published under (GPLv3+).
> 
> Because loudmax's license is incompatible, it cannot be legally included as a part of easyeffects.
> 
> If you really need to use loudmax outside of a DAW, [carla](https://kx.studio/Applications:Carla) is probably what you want.

carla is out of scope, very complicated to use

**Reply by violetmage** [reply]
If the default settings for the limiter were improved, would that alleviate your concern that setup is too complicated?

**Reply by wwmm** [reply]
>none of the existing ones are as easy as this:

I have my doubts if windows is actually doing what that name suggests. It probably is just doing dynamic range compression and not real loudness equalization. In any case in EasyEffects we try to do as much as possible offering as few plugins as possible because the more we add the harder it is to maintain the whole thing. I only made an exception for noise remover plugins because it is a case where some of them are completely unable to remove some kind of noises independently of how much effort is put on configuring them. That is not the case with loudness equalization. It is not going to be an one click solution. But the currently available plugins for loudness equalization and dynamic range compression are enough to get the job done.

**Reply by wwmm** [reply]
> @wwmm
> 
> https://learn.microsoft.com/en-us/windows-hardware/drivers/audio/loudness-equalization-dsp
> 
> i tried to reproduce it, best i could reach: #3800
> 
> you added community pressets, would be nice to have built-in ones too, for things like that

Honestly what they are describing feels similar to what is done by our `autogain` plugin. This plugin does correction based on perceived loudness as calculated by the library 'libebur128`.

About built-in presets we have discussed this a lot in the past and the conclusion was always that it is better to leave these things to the community. In the end the user almost always needs to customize. For example the solution you achieved will depend on the signal level in the compressors input. If the input level is too far from the threshold level set to the compressors the result will not be the same as the one you got on your computer. So the user will have to customize the input level or the threshold. Windows is probably handling this under the hoods. Maybe in a similar way as our autogain does.
